exercise-src/react-native/16-security/01-problem/src/services/auth/AuthProvider.tsx

```diff
@@ -14,10 +14,14 @@ GoogleSignin.configure({
 })
 
 const AuthProvider: FC<{ children: ReactNode }> = ({ children }) => {
-  const [userInfo, setUserInfo] = useState<UserInfo | null>()
+  const [userInfo, setUserInfo] = useState<UserInfo | undefined>()
```

Comment:
TLDR: Leave this at `null` and revert the `isAuthenticated` change below.

If you switch the default to `undefined`, the `isAuthenticated` check at the bottom won't work properly.

We technically have THREE auth states in this app (and all apps, actually):
- User is logged in.
- User is logged out.
- We don't know.

"We don't know." happens when the user was previously logged in, but we're still validating/refreshing their previous session. I do _not_ like to equate this with "logged out" (which is what the `undefined` value above will do).

There is a way to remove that third state, but it'll require a few other changes too. That's what the `setUserInfo(null) or setUserInfo(undefined)?` was supposed to be asking me to decide (but I didn't actually explain that, so no way for you to know).

For now, let's leave this at `null` and handle all 3 states (that's what the old code on line 32 below was doing).